test/functional/blockchain.py

```diff
@@ -44,6 +45,38 @@ def run_test(self):
         self._test_stopatheight()
         assert self.nodes[0].verifychain(4, 0)
 
+    def _test_getblockchaininfo(self):
+        keys = [
+            'bestblockhash',
+            'bip9_softforks',
+            'blocks',
+            'chain',
+            'chainwork',
+            'difficulty',
+            'headers',
+            'mediantime',
+            'pruned',
+            'softforks',
+            'verificationprogress',
+        ]
+        res = self.nodes[0].getblockchaininfo()
+        # should have exact fields
+        assert_equal(sorted(res.keys()), keys)
+
+        # restart node with pruning enabled
+        self.stop_node(0)
```

Comment:
You can speed this test up by starting with pruning the first time, and then restart without pruning (saves one stop-start which is a slow operation). Do that by setting `self.extra_args = [['-stopatheight=207', '-prune=1']]` in `set_test_params()` and then setting `self.extra_args = [['-stopatheight=207']]` before restarting.